bridge/slack/handlers.go

```diff
@@ -90,248 +98,177 @@ func (b *Bslack) handleMatterHook(messages chan *config.Message) {
+		(len(ev.Attachments) > 0 && ev.Attachments[0].CallbackID == "matterbridge_"+b.uuid) {
+		return true
 	}
-	// add the downloaded data to the message
-	helper.HandleDownloadData(b.Log, rmsg, file.Name, comment, file.URLPrivateDownload, data, b.General)
-	return nil
-}
 
-// handleUploadFile handles native upload of files
-func (b *Bslack) handleUploadFile(msg *config.Message, channelID string) {
-	for _, f := range msg.Extra["file"] {
-		fi := f.(config.FileInfo)
-		if msg.Text == fi.Comment {
-			msg.Text = ""
-		}
-		/* because the result of the UploadFile is slower than the MessageEvent from slack
-		we can't match on the file ID yet, so we have to match on the filename too
-		*/
-		b.Log.Debugf("Adding file %s to cache %s", fi.Name, time.Now().String())
-		b.cache.Add("filename"+fi.Name, time.Now())
-		res, err := b.sc.UploadFile(slack.FileUploadParameters{
-			Reader:         bytes.NewReader(*fi.Data),
-			Filename:       fi.Name,
-			Channels:       []string{channelID},
-			InitialComment: fi.Comment,
-		})
-		if res.ID != "" {
-			b.Log.Debugf("Adding fileid %s to cache %s", res.ID, time.Now().String())
-			b.cache.Add("file"+res.ID, time.Now())
-		}
-		if err != nil {
-			b.Log.Errorf("uploadfile %#v", err)
-		}
+	// It seems ev.SubMessage.Edited == nil when slack unfurls.
+	// Do not forward these messages. See Github issue #266.
+	if ev.SubMessage != nil &&
+		ev.SubMessage.ThreadTimestamp != ev.SubMessage.Timestamp &&
+		ev.SubMessage.Edited == nil {
+		return true
 	}
+	return false
 }
 
-// handleMessageEvent handles the message events
+// handleMessageEvent handles the message events. Together with any called sub-methods,
+// this method implements the following event processing pipeline:
+//
+// 1. Check if the message should be ignored.
+//    NOTE: This is not actually part of the method below but is done just before it
+//          is called via the 'skipMessageEvent()' method.
+// 2. Populate the Matterbridge message that will be sent to the router based on the
+//    received event and logic that is common to all events that are not skipped.
+// 3. Detect and handle any message that is "status" related (think join channel, etc.).
+//    This might result in an early exit from the pipeline and passing of the
+//    pre-populated message to the Matterbridge router.
+// 4. Handle the specific case of messages that edit existing messages depending on
+//    configuration.
+// 5. Handle any attachments of the received event.
+// 6. Check that the Matterbridge message that we end up with after at the end of the
+//    pipeline is valid before sending it to the Matterbridge router.
 func (b *Bslack) handleMessageEvent(ev *slack.MessageEvent) (*config.Message, error) {
-	var err error
-
-	// update the userlist on a channel_join
-	if ev.SubType == sChannelJoin {
-		b.populateUsers()
-	}
-
-	// Edit message
-	if !b.GetBool(editDisableConfig) && ev.SubMessage != nil && ev.SubMessage.ThreadTimestamp != ev.SubMessage.Timestamp {
```

Comment:
Any reason why you removed this `!b.GetBool(editDisableConfig)` in your diff ? We can drop this here if we aren't going to do anything with edits

Reply:
The `not been dropped` link doesn't point to anything :(

Reply:
Sorry don't know why it messed up... basically it's at line 154 in the new version of the file.